### ISSUE TITLE
ParserError: expected '<document start>', but found '<scalar>'

### ISSUE THREAD

**Issue by DemGiran** [Battleman/zoomdl#22]
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "./zoomdl/__main__.py", line 20, in <module>
  File "./zoomdl/zoom_dl/__init__.py", line 18, in main
  File "./zoomdl/zoom_dl/zoomdl.py", line 154, in download
  File "./zoomdl/zoom_dl/zoomdl.py", line 76, in get_page_meta
  File "/usr/local/lib/python3.8/site-packages/yaml/__init__.py", line 142, in full_load
    return load(stream, FullLoader)
  File "/usr/local/lib/python3.8/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/usr/local/lib/python3.8/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/usr/local/lib/python3.8/site-packages/yaml/composer.py", line 39, in get_single_node
    if not self.check_event(StreamEndEvent):
  File "/usr/local/lib/python3.8/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/local/lib/python3.8/site-packages/yaml/parser.py", line 171, in parse_document_start
    raise ParserError(None, None,
yaml.parser.ParserError: expected '<document start>', but found '<scalar>'
  in "<unicode string>", line 45, column 2:
    };
     ^

**Comment by Battleman** [comment]
Thanks for the feedback. Please edit your issue so it matches the template for a bug. It contains useful information for debug.
And if you'd agree to share the URL (here, or in PM in [reddit](https://reddit.com/u/gsbattleman)), it would help immensly. I can't reproduce the bug for now.

**Comment by Gh0stByte** [comment]
> 
> 
> Thanks for the feedback. Please edit your issue so it matches the template for a bug. It contains useful information for debug.
> And if you'd agree to share the URL (here, or in PM in [reddit](https://reddit.com/u/gsbattleman)), it would help immensly. I can't reproduce the bug for now.

I have a URL which does the same thing after authentication. Sent you a PM on reddit.

**Comment by automatata** [comment]
I also encounter the same issue. Tried the older v2020.07.07 and it was fine. Just for the record. Not sure for your URLs.

**Comment by Battleman** [comment]
OK, I figured it out. The problem is a faulty regex ([here](https://github.com/Battleman/zoomdl/blob/8db2af1d6b02b36a90754c82682ed44738cc98df/zoom_dl/zoomdl.py#L73)). When there is a chat history, my regex captures more than necessary, and then yaml fails to understand it.
I fixed it (literally only added a semicolon), and made it more robust with a try/except. Will be rolling in 2020.09.14, later today. Need to re-generate exe, and I don't have my Windows VM at hand.
